vendor/github.com/openshift/api/operator/v1/types_etcd.go

```diff
@@ -43,6 +43,18 @@ type EtcdSpec struct {
 	// +openshift:enable:FeatureGate=HardwareSpeed
 	// +optional
 	HardwareSpeed ControlPlaneHardwareSpeed `json:"controlPlaneHardwareSpeed"`
+
+	// backendQuotaGiB sets the etcd backend storage size limit in gibibytes.
+	// The value should be an integer not less than 8 and not more than 32.
+	// When not specified, the default value is 8.
+	// +kubebuilder:default:=8
+	// +kubebuilder:validation:Minimum=8
+	// +kubebuilder:validation:Maximum=32
+	// +kubebuilder:validation:XValidation:rule="self>=oldSelf",message="etcd backendQuotaGiB may not be decreased"
+	// +openshift:enable:FeatureGate=EtcdBackendQuota
```

Comment:
don't we have to check the feature gates as well? 

similar to how we did it for the backups:
https://github.com/openshift/cluster-etcd-operator/blob/master/pkg/operator/starter.go#L451-L454

Reply:
yes i missed that, thanks 👍🏽

Reply:
I actually dont think we need to add this in the `starter.go` since we dont not create a separate controller 

However, I have added these check in the `EnvVarController`, if the featuregate is not enabled, then we dont check the field within the CR, as it does not exist on the CRD on a cluster without this feature gate enabled 

please correct me if i am wrong

Reply:
sounds good!